### PR TITLE
Modfiy the circleci build process so that the build will fail if the git tag and setup.py do not match

### PR DESCRIPTION
**Pull request by chityala-ion**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -87,13 +87,37 @@ jobs:
...
+    docker:
+      - image: circleci/python:3.7 # container for the build job
 
+    steps:
+      - checkout:
+          path: ~/mibitracker-client
+      - run:
+          name: Check if tag and setup.py are consistent
+          command: |
+            python check_tag.py
```
Comment (murataksoy) [inline .circleci/config.yml]: I assume you confirmed that this works without any python environment or additional libraries, correct?

File: check_tag.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, _ = proc.communicate()
+    version = stdout.strip().decode('utf-8')
+    pattern = r"v(\d+).(\d+).(\d+)(.*?)"
+    match = re.match(pattern, version)
+    if match:
+        major = match.group(1)
+        minor = match.group(2)
+        build = match.group(3)
+        return "v"+".".join((major, minor, build))
```
Comment (murataksoy) [inline check_tag.py]: For this and the next function, why do you need to split into minor-major-build? Don't you want to check check if the tag names are the same, regardless of their format ?

File: check_tag.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    major = match.group(1)
+                    minor = match.group(2)
+                    build = match.group(3)
+                    return "v"+".".join((major, minor, build))
+    return ""
+
+if __name__ == '__main__':
+    if not check_tag():
+        raise \
+            Exception("The version number in setup.py and git tag do not match")
```
Comment (murataksoy) [inline check_tag.py]: Can you also write some extra comments to guide the user in this case, e.g., "please change bla bla line in setup.py to -print current version here- " ?

File: .circleci/config.yml (hunk 1 above)
Comment (chityala-ion) [reply .circleci/config.yml]: Yes. It works on github.

File: check_tag.py (hunk 3 above)
Comment (chityala-ion) [reply check_tag.py]: That is a good idea. Will do.

File: check_tag.py (hunk 2 above)
Comment (chityala-ion) [reply check_tag.py]: I initially wanted to get the three numbers and compare them. Finally settled down to comparing the string. I will change it.

**Pull request merged**